### PR TITLE
Fix numerous bugs and issues in FTP over TLS for ftp_list and ftp_download

### PR DESCRIPTION
**Pull request by dprus**
Fix some bugs that broke FTP over TLS as well as bugs that broke ftp_list and ftp_download when file and folder names had special characters. Added better logging in some instances to help diagnose issues. Also fixed a bug where the original coder was incorrectly calling entry.failed(...) instead of entry.fail(...).

### CODE REVIEW

File: flexget/plugins/input/ftp_list.py [hunk 1]
```diff
@@ -143,8 +148,7 @@ def _handle_path(self, entries, ftp, baseurl, path='', mlst_supported=False, fil
                                   recursive, get_size, encoding)
 
             if not files_only or mlst.get('type') == 'file':
-                url = baseurl + path + '/' + p
-                url = url.replace(' ', '%20')
+                url = baseurl + urllib.quote_plus(path) + '/' + urllib.quote_plus(p)
```
Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: Won't this also quote path separators (`/`) in `path`? It shouldn't do that, right?

Comment (dprus) [inline flexget/plugins/input/ftp_list.py]: It doesn't. Here's the python documentation: https://docs.python.org/2/library/urllib.html
As you can see the path separator is considered one of the default "safe" characters that does not get converted.

Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: > It also does not have safe default to '/'.

From said docs.

Comment (dprus) [inline flexget/plugins/input/ftp_list.py]: Ah yes, you're right. I guess the reason it worked was that I use the unquote_plus option in ftp_download. I'm actually a little confused as to what the correct behavior should be. i.e. Since my fix works with the ftp_list and ftp_download input+output combination I assumed I was good. However, if a different combination was used it might not work. Would the correct behavior be for me to add the path separator to the safe parameter in the call?

Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: I'm not 100% sure, but that seems like a reasonable solution to me.

Comment (dprus) [inline flexget/plugins/input/ftp_list.py]: OK, I'll go ahead and cancel this merge request, make the update and resubmit. Thanks.

Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: You can just add another commit to this branch, and the PR will update.

Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: And maybe see how a browser handles the generated url for a general test if it's the right thing to do?

Comment (dprus) [inline flexget/plugins/input/ftp_list.py]: OK, the fix was actually a lot easier. I'm just going to use quote(...) instead of quote_plus(...). I also tested the generated URL in Firefox and it works even when I use all sorts of spaces and weird characters in the URL. However, I uncovered an issue with ftp_download in the process. It's not downloading the files to the same folder hierarchy on the local disk. The issue appears to have always been there though? I'm going to try and track it down and fix it.

Comment (gazpachoking) [inline flexget/plugins/input/ftp_list.py]: Okay, let's make that a separate pr though.

**Comment by gazpachoking** [comment]
Thanks!
